### PR TITLE
Fixed an issue with default values of current time arguments

### PR DESCRIPTION
**Pull request by grixa**
Used std::chrono::microseconds(-1) instead of current::time::Now() as default arguments values.
CC @dkorolev @mzhurovich

### CODE REVIEW

File: Blocks/MMQ/mmpq.h [hunk 1]
```diff
@@ -74,8 +74,9 @@ class MMPQ {
   // NOTE(dkorolev): `std::enable_if_t<std::is_same<message_t, current::decay<T>>::value, idxts_t>` can't convert
   // `const char*` into an `std::string`, which is essential, as, unlike MMQ, MMPQ is not an `ss::EntryPublisher<>`.
   template <current::locks::MutexLockStatus MLS = current::locks::MutexLockStatus::NeedToLock, class T>
-  idxts_t Publish(T&& message, std::chrono::microseconds timestamp = current::time::Now()) {
+  idxts_t Publish(T&& message, std::chrono::microseconds us = std::chrono::microseconds(-1)) {
```
Comment (dkorolev) [inline Blocks/MMQ/mmpq.h]: Why not use `std::chrono:microseconds(-1)` as a constant, and compare to it directly, w/o `.count()`?

File: Blocks/MMQ/mmq.h [hunk 2]
```diff
@@ -103,11 +99,6 @@ class MMQImpl {
   // THREAD SAFE. Blocks the calling thread for as short period of time as possible.
   template <current::locks::MutexLockStatus MLS>
   idxts_t DoPublish(const message_t& message, std::chrono::microseconds timestamp) {
-    if (CHECK_TIMESTAMP_MONOTONICITY) {
-      if (!(timestamp > last_idx_ts_.us)) {
-        CURRENT_THROW(ss::InconsistentTimestampException(last_idx_ts_.us + std::chrono::microseconds(1), timestamp));
```
Comment (dkorolev) [inline Blocks/MMQ/mmq.h]: These checks and `CURRENT_THROW`-s are still essential.

File: Blocks/MMQ/mmq.h [hunk 3]
```diff
@@ -201,13 +187,16 @@ class MMQImpl {
...
   template <bool DROP = DROP_ON_OVERFLOW>
-  typename std::enable_if<DROP, std::pair<bool, size_t>>::type CircularBufferAllocate(
-      std::chrono::microseconds timestamp) {
+  typename std::enable_if<DROP, std::pair<bool, size_t>>::type CircularBufferAllocate(std::chrono::microseconds us) {
     // Implementation that discards the message if the queue is full.
     // MUTEX-LOCKED.
     std::lock_guard<std::mutex> lock(mutex_);
     if (circular_buffer_[head_].status == Entry::FREE) {
       // Regular case.
+      const auto timestamp = us.count() >= 0 ? us : current::time::Now();
```
Comment (dkorolev) [inline Blocks/MMQ/mmq.h]: If in this method there is no default parameter, perhaps just `assert` `us.count() >= 0`?

File: Blocks/MMQ/mmq.h (hunk 2 above)
Comment (grixa) [inline Blocks/MMQ/mmq.h]: They are inside the `CircularBufferAllocate` now.

File: Blocks/MMQ/mmq.h (hunk 3 above)
Comment (grixa) [inline Blocks/MMQ/mmq.h]: Yes, there is no default parameter. But it doesn't mean the `CircularBufferAllocate` couldn't be called with `us == std::chrono::microseconds(-1)`, it just mean no one needs to call it without specifying `us` value explicitly. The default `-1` value from `EntryPersister::Publish` would be passed here through the `MMQImpl::DoPublish` without modification. And here, at last, we can call `Now()`.

**Pull request merged**